compiler/modules/front.syntax/token.bal

```diff
@@ -487,9 +487,70 @@ public readonly class SourceFile {
         return unpackPosition(pos);
     }
 
-    public function lineContent(Position pos) returns string {
-        int lineNum = self.lineColumn(pos)[0];
-        return scanLineToString(self.scannedLine(lineNum));
+    // range is expected to;the start of a token
+    public function lineContent((Position|d:Range) range) returns [string, string, string] {
```

Comment:
You should be able to implement this in one iteration over the ScannedLine.  I don't think scanLineToString and scanLineFragIndex are the right primitives for this.

This needs more thought...

Reply:
fixed